explainer.md

```diff
@@ -0,0 +1,166 @@
+- If there are existing features that conflict with the front-facing camera feature, the WebXR Device API already provides a mechanism to resolve this. Unsatisfiable set of required features will be rejected, and an unsatisfiable set of optional features should cause the user agent to still create a session, albeit with unspecified features (which the site should tolerate, since it listed those features as not critical to the experience).
+
+The above discussion also serves as the argument for treating `“front-facing”` as a new feature descriptor (as opposed to a configuration knob), as we want it to be able to participate in the feature resolution algorithm. Were it a configuration knob (either exposed at session creation, or afterwards), it could force the site to re-try session creation attempt if front-facing camera is only optionally needed (example #4 from [above](note-on-session-creation) would not be expressible).
+
+Open question: should we also add a new feature descriptor for back-facing camera? Given that there is an existing feature incubation for expression tracking, which could theoretically be implemented for smartphone AR in front-facing camera mode, the sites could want to force the use of an environment-facing camera. Alternatives would be to never use a front-facing camera when the sites didn’t explicitly ask for it via the `”front-facing”` feature descriptor (i.e. we won’t be able to add a feature that’d imply the use of front-facing camera), or treat `”front-facing”` feature descriptor as implicitly present when requesting such features. Implicitly adding the feature descriptor comes with its own set of problems though (i.e. it’d force all the sites to always check whether they use the camera they needed, which could be a back-compat issue).
+
+#### Simultaneous access to cameras
+
+In the future, it may be possible to leverage multiple cameras to power the AR sessions on smartphones, and the underlying AR frameworks may make it possible to access the data coming from multiple cameras. The API shape proposed above should at the very least not conflict with this capability. One way to implement simultaneous access to cameras would be to treat the `”front-facing”` feature descriptor as a modifier that only influences the camera that will be selected for the immersive experience (i.e. camera selection for primary views), and treat other camera streams as secondary views, which should already be covered by `”secondary-views”` [feature](https://www.w3.org/TR/webxr/#secondary-view-secondary-views).
+
+#### Alternative #1 - camera facing is a configuration knob
+
+```webidl
+enum XRCameraFacing {
+  "back",
+  "front",
+};
+
+dictionary XRCameraConfigInit {
+  optional XRCameraFacing facing = "back",
+  optional XRCameraDisplayOrientation displayOrientation = "mirrored",
```

Comment:
This alternative mirrors the 8th wall APIs. They theoretically allow you to mirror the back camera, and needs opt in to flip on the front facing. 

Although mirroring the back camera isn't that useful to users in general, it can be useful for decoupling front/back intrinsics/extrinsics transforms and mirrored/non-mirrored intrinsics/extrinsics transforms that can help simplify (and test) the problem.